crypto/ec/ec_lib.c

```diff
@@ -621,9 +626,7 @@ int EC_POINT_copy(EC_POINT *dest, const EC_POINT *src)
         return 0;
     }
     if (dest->meth != src->meth
-            || (dest->curve_name != src->curve_name
-                && dest->curve_name != 0
-                && src->curve_name != 0)) {
```

Comment:
So the curve name test may be an issue (i.e. Do multiple nids map to the same curve, and would there be some issues with some of prime curves that have multiple implementations).
My thought was that if they use the same point comparison method then they should be able to be compared. I am not convinced that a different method table equates to every method in the table being incompatible for all operations. It would of been better to have some sort of type(s) to indicate if methods are incompatible.

Reply:
Looks like there are multiple curve names for the same data e.g:
 {NID_wap_wsg_idm_ecid_wtls3, &_EC_NIST_CHAR2_163K.h, 0,
{NID_sect163k1, &_EC_NIST_CHAR2_163K.h, 0,

Reply:
Changed it back until I can reproduce what the problem was. ec_point_is_compat(0 is not able to be used as its a point, group test.